tests/test_flow/test_persistence_random.py

```diff
@@ -194,21 +259,77 @@ class ModelBinding:
     dep_entity_names = attr.ib()
     use_tuples_for_output = attr.ib()
     func_version = attr.ib(default=1)
+    annotated_func_version = attr.ib(default=1)
 
     has_persisted_values = attr.ib(default=False)
     persisted_values_by_entity_name = attr.ib(default=None)
+    persisted_values_stale_ancestor = attr.ib(default=None)
+
     some_descendant_might_have_persisted_values = attr.ib(default=False)
+    some_descendant_might_have_fresh_persisted_values = attr.ib(default=False)
```

Comment:
I think it's clearer to say `stale_persisted_values` and flip the flag. The names are getting wordy but I don't have a better suggestion either.

Reply:
So you'd call it `no_descendant_can_have_stale_persisted_values`? In that case I'd probably want to flip both of them; I'll try that and see if it looks better.